Bigtable/tests/Unit/InstanceTest.php

```diff
@@ -0,0 +1,214 @@
+    }
+
+    public function testCreateWithoutbuildClusterMetadata()
+    {
+        try {
+            $this->instance->create(
+                []
+            );
+        }  catch(\Exception $e) {
+            $error = 'At least one clusterMetadata must be passed';
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreateWithoutClusterId()
+    {
+        try {
+            $clusterMetadataList = $this->bigtableClient->buildClusterMetadata(null, null);
+            $this->instance->create(
+                [$clusterMetadataList]
+            );
+        }  catch(\Exception $e) {
+            $error = 'Cluster id must be set';
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreateWithClusterIdBadFormat()
+    {
+        try {
+            $badClusterId = 'badformat/my-cluster';
+            $clusterMetadataList = $this->bigtableClient->buildClusterMetadata($badClusterId, 'location-id');
+            $this->instance->create(
+                [$clusterMetadataList]
+            );
+        }  catch(\Exception $e) {
+            $error = "Please pass just clusterId as 'cluster-id'";
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreateWithoutLocationId()
+    {
+        try {
+            $clusterMetadataList = $this->bigtableClient->buildClusterMetadata(self::CLUSTER_ID, null);
+            $this->instance->create(
+                [$clusterMetadataList]
+            );
+        }  catch(\Exception $e) {
+            $error = 'Location id must be set';
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreateWithLocationIdBadFormat()
+    {
+        $badLocationId = 'badformat/my-locations';
+        $clusterMetadataList = $this->bigtableClient->buildClusterMetadata(
+            self::CLUSTER_ID,
+            $badLocationId
+        );
+        try {
+            $this->instance->create(
+                [$clusterMetadataList]
+            );
+        }  catch(\Exception $e) {
+            $error = "Please pass just locationId as 'location-id'";
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreateWithServeNodeIsZero()
+    {
+        $clusterMetadataList = $this->bigtableClient->buildClusterMetadata(
+            self::CLUSTER_ID,
+            self::LOCATION_ID,
+            null,
+            0
+        );
+        try {
+            $this->instance->create(
+                [$clusterMetadataList]
+            );
+        }  catch(\Exception $e) {
+            $error = "When creating Production instance, serveNodes must be > 0";
+            $this->assertEquals($error, $e->getMessage());
+        }
+    }
+
+    public function testCreate()
+    {
+        $this->connection->createInstance(Argument::any())
```

Comment:
In unit tests, you should NOT use `Argument::any()`. The point of the unit test is to verify that you are handling input properly and receiving the correct output. You've got the output covered, but not input.

You can simply provide an array exactly matching the expected input, or if there are only limited things you need to test, you can use the [Argument tokens](https://github.com/phpspec/prophecy/blob/master/src/Prophecy/Argument.php).

In this case, please provide exact input matching. The example below isn't meant to be a copy-paste (because i'm not sure what the expected value is), but it should demonstrate what I mean in general terms.

```php
$this->connection->createInstance([
    'instanceName' => '<instance-name>',
    'displayName' => 'My instance',
])->shouldBeCalled()->willReturn([...]);
```